src/Streamly/Internal/Memory/Array/Types.hs

```diff
@@ -15,28 +15,33 @@ module Streamly.Internal.Memory.Array.Types
     (
       Array (..)
 
+    -- * Freezing and Thawing
+    , unsafeFreeze
+    , unsafeFreezeWithShrink
+    , unsafeThaw
+
     -- * Construction
-    , withNewArray
-    , newArray
-    , unsafeSnoc
-    , snoc
-    , spliceWithDoubling
+--    , withNewArray
+--    , newArray
+--    , unsafeSnoc
+--    , snoc
+--    , spliceWithDoubling
     , spliceTwo
 
-    , fromAddr#
-    , fromString#
+--    , fromAddr#
+--    , fromString#
     , fromList
     , fromListN
     , fromStreamDN
     -- , fromStreamD
 
     -- * Streams of arrays
     , fromStreamDArraysOf
-    , FlattenState (..) -- for inspection testing
+    , MA.FlattenState (..) -- for inspection testing
```

Comment:
Why are we exporting Mutable array stuff from immutable array?

Reply:
Yeah, I should remove these.